vignettes/adtr.Rmd

```diff
@@ -0,0 +1,599 @@
+flag variables that will then aid setting up any additional analysis flag
+variables defined as per your company specifications.
+
+###  Derive `CRFL`, `CRNFL`
+
+In this example, `CRFL` flags where the patient has had Complete Response, and
+`CRNFL`, flags where Complete response contributed to the NADIR calculation.
+
+```{r}
+adtr_sum <- adtr_sum %>% mutate(
+  CRFL = if_else(AVAL == 0 & ANL01FL == "Y", "Y", NA_character_),
+  CRNFL = if_else(NADIR == 0, "Y", NA_character_)
+)
+```
+
+### Derive `PDFL`
+
+In this example, we want to flag when a patient has had PD.  This could be done
+in a number of ways:
+
+1. take the date of PD from `ADRS`
+
+2. take the first date when the overall response is PD from `RS` or
+
+3. Calculate from source data, the derivation for PD for target lesions is based
+on the calculated sum of diameters of the assessed lesions results in a
+calculated Percent Change from NADIR >=20% and a calculated Change from NADIR
+>=5mm depending on your company specifications.
+
+See below example.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  mutate(
+    PDFL = if_else(PCHGNAD >= 20 & CHGNAD >= 5 | is.na(CRFL) & CRNFL == "Y", "Y", NA_character_)
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, CHGNAD, PCHGNAD, PDFL)
+)
+```  
+
+## Derive Analysis Flags `ANL02FL`, `ANL03FL`, `ANL04FL` {#analflag}
+
+Based on your analysis, additional analysis flags could be set up in `ADTR`.
+Below are some examples
+
+Analysis 02 Flag only includes the visit with the lowest NADIR (lowest sum of
+diameter) compared across all visits.  Use PCHGNNAD or PCHG?
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  restrict_derivation(
+    derivation = derive_var_extreme_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      new_var = ANL02FL,
+      order = vars(AVAL),
+      mode = "first"
+    ),
+    filter = ANL01FL == "Y"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, ANL02FL)
+)
+```  
+
+Analysis 03 flag includes `ANL01FL="Y"` and the target lesion sum of diameter
+until patient has had PD. Use the `PDFL` variable.
+
+```{r}
+adtr_sum <- adtr_sum %>%
+  restrict_derivation(
+    derivation = derive_var_relative_flag,
+    args = params(
+      by_vars = vars(USUBJID),
+      new_var = ANL03FL,
+      condition = PDFL == "Y",
+      order = vars(ADY),
+      mode = "first",
+      selection = "before",
+      inclusive = TRUE
+    ),
+    filter = ANL01FL == "Y"
+  )
+```
+
+```{r, echo=FALSE}
+dataset_vignette(
+  arrange(adtr_sum, USUBJID, AVISITN),
+  display_vars = vars(USUBJID, AVISIT, AVAL, NADIR, PDFL, ANL03FL)
+)
+```  
+
+Analysis 04 flag includes all sum of diameters even where lesion identifiers
+match and unmatch if the patient has had PD. Use `CRFL`, `CRNFL`, `PDFL` variables.
```

Comment:
I think we do not need `CRFL` and `CRNFL` here. (They were already considered for `PDFL`.)

Reply:
we don't use them in our reporting , so its fine to remove CRFL,CRNFL

Reply:
we are keeping them finally. @bundfussr I okay to close this comment, if you are okay , can you close it